test/units/module_utils/common/text/test_converters.py

```diff
@@ -0,0 +1,245 @@
+            (1, 2),
+        ]
+    )
+    def test_default_behavior(self, test_input):
+        """
+        Test for _json_encode_fallback() using other types.
+
+        It must fail with TypeError.
+        """
+
+        with pytest.raises(TypeError, match='Cannot json serialize'):
+            _json_encode_fallback(test_input)
+
+
+class TestContainerToBytes:
+
+    """Namespace for testing container_to_bytes()."""
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (b'str', b'str'),
+            (u'str', b'str'),
+            ([u'str'], [b'str']),
+            ((u'str'), (b'str')),
+            ({u'str': u'str'}, {b'str': b'str'}),
+        ]
+    )
+    @pytest.mark.parametrize('encoding', ['utf-8', 'latin1', 'shift_jis', 'big5', 'koi8_r'])
+    @pytest.mark.parametrize('errors', ['strict', 'surrogate_or_strict', 'surrogate_then_replace'])
+    def test_to_bytes_different_types(self, test_input, expected, encoding, errors):
+        """Test for passing objects to container_to_bytes()."""
+
+        assert container_to_bytes(test_input, encoding=encoding, errors=errors) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (True, True),
+            (None, None),
+            (u'str', u'str'.encode(DEFAULT_ENCODING)),
+            (u'くらとみ', u'くらとみ'.encode(DEFAULT_ENCODING)),
+            (u'café', u'café'.encode(DEFAULT_ENCODING)),
+            (b'str', u'str'.encode(DEFAULT_ENCODING)),
+            (u'str', u'str'.encode(DEFAULT_ENCODING)),
+            ([u'str'], [u'str'.encode(DEFAULT_ENCODING)]),
+            ((u'str'), (u'str'.encode(DEFAULT_ENCODING))),
+            ({u'str': u'str'}, {u'str'.encode(DEFAULT_ENCODING): u'str'.encode(DEFAULT_ENCODING)}),
+        ]
+    )
+    def test_to_bytes_default_encoding_err(self, test_input, expected):
+        """
+        Test for passing objects to container_to_bytes(). Default encoding and errors
+        """
+
+        assert container_to_bytes(test_input, encoding=DEFAULT_ENCODING,
+                                  errors=DEFAULT_ERR_HANDLER) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,encoding,expected',
+        [
+            (u'くらとみ', 'latin1', b'????'),
+            (u'café', 'shift_jis', b'caf?'),
+        ]
+    )
+    def test_incompatible_chars_and_encodings(self, test_input, encoding, expected):
+        """
+        Test for passing incompatible characters and encodings container_to_bytes().
+        """
+
+        # tests for surrogate_or_strict and strict:
+        for err in ('surrogate_or_strict', 'strict'):
+            with pytest.raises(UnicodeEncodeError, match="codec can't encode"):
```

Comment:
You're checking two separate properties here. This should be in a separate test.